server/lib/redis.ts

```diff
@@ -24,11 +24,15 @@ class Redis {
     if (this.initialized === true) return
     this.initialized = true
 
-    this.client = createClient({
-      host: CONFIG.REDIS.HOSTNAME,
-      port: CONFIG.REDIS.PORT,
-      db: CONFIG.REDIS.DB
-    })
+    this.client = createClient(
+      Object.assign({},
+        (CONFIG.REDIS.AUTH && CONFIG.REDIS.AUTH != null) ? { password: CONFIG.REDIS.AUTH } : {},
+        (CONFIG.REDIS.DB) ? { db: CONFIG.REDIS.DB } : {},
+        (CONFIG.REDIS.HOSTNAME && CONFIG.REDIS.PORT) ?
+        { host: CONFIG.REDIS.HOSTNAME, port: CONFIG.REDIS.PORT } :
+        { path: CONFIG.REDIS.SOCKET }
```

Comment:
I've never seen Object.assign() used to compose an object like this (steals technique) 👍

Reply:
ah that's much cleaner. Is a static method for in `lib/redis.ts` okay?

Reply:
:+1: